lib/src/combine_latest.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+/// Combine the latest value from the source stream with the latest value from
+/// [combineWith] using [combine].
```

Comment:
nit: add a clarifying note here that this emits a value any time either stream emits a value as opposed to waiting for a new value from each stream before emitting a new value.

I think it might be worth even listing a quick example similar to how the first test looks.

Reply:
Added another sentence and a timeline diagram - does that make it clear?